Polyfills/Window/Source/Window.cpp

```diff
@@ -6,6 +6,112 @@
 #include <chrono>
 #include <iterator>
 
+namespace
+{
+    struct TimeoutFunction
+    {
+        std::shared_ptr<Napi::FunctionReference> Function;
+        std::chrono::time_point<std::chrono::system_clock> TimePoint;
+
+        TimeoutFunction(std::shared_ptr<Napi::FunctionReference> func, std::chrono::time_point<std::chrono::system_clock> time)
+            : Function{std::move(func)}
+            , TimePoint{time}
+        { }
+
+        static bool Compare(const TimeoutFunction& tf1, const TimeoutFunction& tf2)
+        {
+            return tf1.TimePoint > tf2.TimePoint;
+        }
+    };
+
+    class TimeoutDispatcher
+    {
+    public:
+        TimeoutDispatcher(Babylon::JsRuntime& runtime)
+            : m_runtime(runtime)
+            , m_thread{&TimeoutDispatcher::WaitThenCallProc, this}
+        { }
+
+        void Dispatch(std::shared_ptr<Napi::FunctionReference> func, std::chrono::system_clock::time_point time) 
+        {   
+            if (time <= std::chrono::system_clock::now())
```

Comment:
FYI, `system_clock` can roll backwards if the user adjusts their system clock. `steady_clock` is guaranteed to always progress forward. I'd recommend using `steady_clock` for everything here.